pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentSubscription.java

```diff
@@ -291,12 +291,7 @@ public CompletableFuture<Void> addConsumer(Consumer consumer) {
                     }
                 }
 
-                try {
-                    dispatcher.addConsumer(consumer);
```

Comment:
this is great ! the code already expected a CP

Reply:
yes, it fits well into that.